app/lib/features/onboarding/pages/link_email_page.dart

```diff
@@ -105,6 +106,7 @@ class LinkEmailPage extends ConsumerWidget {
     EasyLoading.show(status: L10n.of(context).linkingEmailAddress);
     try {
       await manager.requestTokenViaEmail(emailController.text.trim());
+      ref.invalidate(emailAddressesProvider);
```

Comment:
I believe we were on a way to refactor ```ref.invalidate``` in this specific case and use proper object updates underlying client. Can you brief why it's needed?

Reply:
This is needed. While we do want to get rid of `ref.invalidate` for all data that is coming through `sync`, the email address providing isn't managed through sync but has a manual request scheme it needs. Those need to be actively refreshed unfortunately and that is happening here.